### ISSUE TITLE
Looping gifs with PillowWriter

### ISSUE THREAD

**Issue by t-makaro** [matplotlib/matplotlib#11787]
### Bug report

**Bug summary**

Saving a gif using PillowWriter produces a gif that does not loop. Is there a way to make the gif loop endlessly using PillowWriter?

**Code for reproduction**

```python
from matplotlib.animation import FuncAnimation, PillowWriter
import matplotlib.pyplot as plt
import numpy as np

x = np.linspace(0, 1, 20)

def f(x, i):
    return x*np.sin(i/10)

line, = plt.plot(x, f(x, 0))
plt.xlim([0,1])
plt.ylim([-1,1])
fig = plt.gcf()
def animate(i):
    line.set_data(x, f(x,i))

anim = FuncAnimation(fig, animate, frames=60, interval=50)
anim.save('fig.gif', writer=PillowWriter(fps=20))
plt.show()
```

**Actual outcome**

![fig](https://user-images.githubusercontent.com/18018386/43351385-004606d0-91c6-11e8-9b78-12fa10a307f4.gif)

**Expected outcome**

Same gif, but I want it to loop once it finishes.

**Matplotlib version**

  * Operating system: Windows 10 Version 1803
  * Matplotlib version: 2.2.2
  * Matplotlib backend (`print(matplotlib.get_backend())`): Qt5Agg
  * Python version: 3.6.5

Matplotlib was installed with conda.

**Comment by dopplershift** [comment]
If someone knows the incantation to use with `pillow` to make a gif loop, then we can add it.

**Comment by t-makaro** [comment]
I found it. According to the [Pillow Docs](https://pillow.readthedocs.io/en/latest/handbook/image-file-formats.html#gif), adding the ```loop``` parameter to the save call to sets the number of times the gif loops. 

I found [this](https://github.com/ManiacalLabs/BiblioPixel/issues/872), and he set ```loop=0```. One might think this that would be not looping (I guess 1 is not looping), but I decided to try it. I simply added ```loop=0``` to the call on [this line](https://github.com/matplotlib/matplotlib/blob/300e4c9afbc9dc00800f36184814634694171e3f/lib/matplotlib/animation.py#L583) (I literally just modified that line in my site-packages), and lone behold endless looping. 

Presumably, we should make this configurable.
```
self.loop = kwargs.pop("loop", 0)
```
to the ```__init__``` of PillowWriter, and then ```loop=self.loop``` to the call to save.
